routing/transit_graph.cpp

```diff
@@ -58,13 +58,31 @@ RouteWeight TransitGraph::CalcSegmentWeight(Segment const & segment,
                      0 /* nontransitCross */);
 }
 
+RouteWeight TransitGraph::GetTransferPenalty(Segment const & from, Segment const & to) const
+{
+  if (!IsEdge(from) || !IsEdge(to))
+    return RouteWeight(0 /* weight */, 0 /* nontransitCross */);
+
+  auto lineIdFrom = GetEdge(from).GetLineId();
+  auto lineIdTo = GetEdge(to).GetLineId();
+
+  if (lineIdFrom == lineIdTo)
+    return RouteWeight(0 /* weight */, 0 /* nontransitCross */);
+
+  auto const it = m_transferPenalties.find(lineIdTo);
+  CHECK(it != m_transferPenalties.cend(), ("Segment", to, "belongs to unknown line", lineIdTo));
+  return RouteWeight(it->second, 0 /* nontransitCross */);
+}
+
 void TransitGraph::GetTransitEdges(Segment const & segment, bool isOutgoing,
                                    vector<SegmentEdge> & edges,
                                    EdgeEstimator const & estimator) const
 {
   CHECK(IsTransitSegment(segment), ("Nontransit segment passed to TransitGraph."));
   for (auto const & s : m_fake.GetEdges(segment, isOutgoing))
-    edges.emplace_back(s, CalcSegmentWeight(isOutgoing ? s : segment, estimator));
+    edges.emplace_back(s,
+                       CalcSegmentWeight(isOutgoing ? s : segment, estimator) +
+                           GetTransferPenalty(isOutgoing ? segment : s, isOutgoing ? s : segment));
```

Comment:
Looks like it's time to separate the branches with an ```if```.

Reply:
``` C++
for (auto const & s : m_fake.GetEdges(segment, isOutgoing))
{
  auto const & from = isOutgoing ? segment : s;
  auto const & to = isOutgoing ? s : segment;
  edges.emplace_back(s, CalcSegmentWeight(to, estimator) + GetTransferPenalty(from, to));
}
```
ok?

Reply:
ok